numpy/linalg/umath_linalg.c.src

```diff
@@ -1092,7 +1092,10 @@ static inline void
     fortran_int info = 0;
     int i;
     /* note: done in place */
-    LAPACK(@cblas_type@getrf)(&m, &m, (void *)src, &m, pivots, &info);
+    if (m != 0) {
+        /* Lapack (at least some versions) do not handle n == 0 */
+        LAPACK(@cblas_type@getrf)(&m, &m, (void *)src, &m, pivots, &info);
```

Comment:
You'd need to look at the documentation for each lapacj function in turn, which can be found within f2c_[czds]_lapack.c